components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -268,6 +268,20 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             }
         }
 
+        String loginHint = request.getParameter(BasicAuthenticatorConstants.LOGIN_HINT);
+
+        if (StringUtils.isNotBlank(loginHint)) {
+            String inputType = FrameworkConstants.INPUT_TYPE_LOGIN_HINT;
+            if (StringUtils.isNotBlank(inputType)) {
+                queryParams += "&" + FrameworkConstants.RequestParams.INPUT_TYPE + "=" + inputType;
+            }
+            context.addEndpointParam(FrameworkConstants.JSAttributes.JS_OPTIONS_USERNAME, loginHint);
+            String additionalParams = runtimeParams.get(ADDITIONAL_QUERY_PARAMS);
```

Comment:
wouldn't this cause any NPE?

Reply:
The runtimeParams can't be null because they are resolved through https://github.com/wso2/carbon-identity-framework/blob/09fd6caaaf37ffb4a1253e6060f18bf8801fea97/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java#L437 this method and it always returns a Map<String, String> eventhough the map could be of size 0.

Reply:
The runtimeParams can't be null because they are resolved through https://github.com/wso2/carbon-identity-framework/blob/09fd6caaaf37ffb4a1253e6060f18bf8801fea97/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java#L437 this method and it always returns a Map<String, String> eventhough the map could be of size 0.

Reply:
The getRuntimeParams is used to set run time parameters. This method doesn't return null, although maps of size zero are possible.  
https://github.com/wso2/carbon-identity-framework/blob/09fd6caaaf37ffb4a1253e6060f18bf8801fea97/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java#L437